### PR TITLE
Allow cluster to be registered in a different project

### PR DESCRIPTION
**Pull request by jasonwzm**
/cc @chizhg

### CODE REVIEW

File: scripts/asm-installer/install_asm [hunk 1]
```diff
@@ -1735,28 +1735,38 @@ get_cluster_labels() {
...
-  if [[ -z "${IDENTITY_PROVIDER}" ]] || [[ "${IDENTITY_PROVIDER}" = 'null' ]]; then
+  if [[ -z "${IDENTITY_PROVIDER}" ]] || [[ "${IDENTITY_PROVIDER}" == 'null' ]]; then
     false
   fi
+
+  populate_environ_info
+
+  if [[ "${PROJECT_ID}" == "${ENVIRON_PROJECT_ID}" ]]; then
+    local WANT
+    WANT="//container.googleapis.com/projects/${PROJECT_ID}/locations/${CLUSTER_LOCATION}/clusters/${CLUSTER_NAME}"
```
Comment (zerobfd) [inline scripts/asm-installer/install_asm:1755]: Can you just replace PROJECT_ID with ENVIRON_PROJECT_ID and remove the `--project "${PROJECT_ID}"` flag?

Comment (jasonwzm) [reply scripts/asm-installer/install_asm:1755]: How about permission check? Try to check membership, if permission issue, then warn?

Comment (zerobfd) [reply scripts/asm-installer/install_asm:1755]: Makes sense to me. We can just make it a general "Failed to validate environ registration" warning.

Comment (jasonwzm) [reply scripts/asm-installer/install_asm]: done.

**Review by zerobfd — APPROVED** [review]

**Pull request merged**